editor/header/header-toolbar/index.js

```diff
@@ -35,6 +36,7 @@ function HeaderToolbar( { hasUndo, hasRedo, hasFixedToolbar, undo, redo } ) {
 				label={ __( 'Redo' ) }
 				disabled={ ! hasRedo }
 				onClick={ redo } />
+			{ isMultiBlockSelection && <BlockSwitcher key="switcher" uids={ selectedBlockUids } /> }
```

Comment:
From a design perspective, this is always showing up in the Header even if I choose to show the block toolbar next to the block, I think we might want to show the switcher in the first selected block's toolbar when this option is toggled. (You can toggle this option in the ellipsis menu at the top right of the editor) cc @jasmussen

Reply:
Regarding this point, I agree, adding the transform drop-down to the tools of the first block may be an option if the config preference is set. But I think adding this logic would increase the scope of this PR and make it harder to review. If it is ok for you I would prefer to add this logic in a future PR.